api/src/main/java/org/openmrs/module/idgen/IdentifierSource.java

```diff
@@ -13,16 +13,18 @@
  */
 package org.openmrs.module.idgen;
 
-import java.util.Date;
-import java.util.Set;
-
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.openmrs.OpenmrsMetadata;
 import org.openmrs.PatientIdentifierType;
 import org.openmrs.User;
 
+import java.util.Date;
+import java.util.Set;
+
 /**
  * An IdentifierSource is a construct which can supply identifiers of a particular type
  */
-public abstract class IdentifierSource {
+public abstract class IdentifierSource implements OpenmrsMetadata {
     
```

Comment:
This is a welcome addition.  At the time this module was written we needed to include support for OpenMRS 1.4 which did not have OpenmrsMetadata available.  However, there are several other methods you should remove in favor of those already on OpenmrsMetadata, notably the Auditable and Retireable fields